src/Synthesiser.cpp

```diff
@@ -2317,17 +2316,18 @@ void Synthesiser::generateCode(std::ostream& os, const std::string& id, bool& wi
 
     os << "if (!opt.parse(argc,argv)) return 1;\n";
 
-    os << "#if defined(_OPENMP) \n";
-    os << "omp_set_nested(true);\n";
-    os << "\n#endif\n";
-
     os << "souffle::";
     if (Global::config().has("profile")) {
         os << classname + " obj(opt.getProfileName());\n";
     } else {
         os << classname + " obj;\n";
     }
 
+    os << "#if defined(_OPENMP) \n";
+    os << "obj.setNumThreads(opt.getNumJobs());\n";
+    os << "omp_set_nested(true);\n";
```

Comment:
Do we need nesting? It may be better without, though this depends on if we do use nesting or not currently.